### ISSUE TITLE
[css-overscroll] the target of overscroll-behavior should be scrolling box instead of scroll container?

### ISSUE THREAD

**Issue by hiikezoe** [w3c/csswg-drafts#6523]
An example;
```
<div style="height: 20vh; overflow: hidden; overscroll-behavior: none;">
  <div>overflow: hidden; overscroll-behavior: none;</div>
  <div style="height: 100vh; width: 100%;"></div>
</div>
<div style="height: 200vh; width: 100%;"></div>
```

In my understandings with the current spec text (i.e. "applied to scroll container"), scrolling on the overflow: hidden element shouldn't be handed off to the outer one, but in fact both Chrome and Firefox hand off it to the outer.

**Comment by hiikezoe** [comment]
When I opened this issue, I was naively thinking all not-scrolling-box should not respect overscroll-behavior properties.  That said, I think now, in the case of viewports things are a bit more complicated.

So for example, if the top level content document specifies `overscroll-behavior: none`, I am pretty sure such sites want to implement their own reload contents feature when the user swipe down the content or something like that. But if the document doesn't have enough contents to make it scrollable (i.e. make it a scrolling box), the site still wants to reload by swiping down? I am pretty sure it does.

Also note that if another website embeds such sites in iframes, users expect the embedded sites not to capture the swipe gestures, instead the users want to scroll the outer document if the embedded documents are not scrollable at all.

**Comment by nanto** [comment]
I think it's helpful if scrolling on an `overflow: auto` element with no scroll range doesn't hand off to outer elements. The major use case is to prevent background scrolling when a modal dialog is open.

Many people are looking for a way to prevent background scrolling when a modal dialog is open. [Some people use `body { overflow: hidden; }`](https://www.folkstalk.com/2022/09/how-to-stop-background-scrolling-when-popup-is-open-with-code-examples.html) but this way has some problems:

1. Needs JavaScript to know the timing to specify `overflow: hidden` to the `body` element.
2. Scrollbar gutter disappears and the reflow happens on the content of the `body` element.

If scrolling on an element with no scroll range doesn't hand off, we can just write as `dialog:modal { overscroll-behavior: contain; }`. [This way](https://jsfiddle.net/6yv230xp/) works as expected on Chrome canary 110 but doesn't work on Firefox nightly 109.

[^1]: e.g.

**Comment by hiikezoe** [comment]
> If scrolling on an element with no scroll range doesn't hand off, we can just write as `dialog:modal { overscroll-behavior: contain; }`. [This way](https://jsfiddle.net/6yv230xp/) works as expected on Chrome canary 110 but doesn't work on Firefox nightly 109.

I tested it on Chrome Canary 110.0.5453.0, it does still hand off.

**Comment by hiikezoe** [comment]
> If scrolling on an element with no scroll range doesn't hand off, we can just write as `dialog:modal { overscroll-behavior: contain; }`. [This way](https://jsfiddle.net/6yv230xp/) works as expected on Chrome canary 110 but doesn't work on Firefox nightly 109.

I found there's [an open spec issue](https://github.com/whatwg/html/issues/7732) for such case.

**Comment by nanto** [comment]
> > If scrolling on an element with no scroll range doesn't hand off, we can just write as `dialog:modal { overscroll-behavior: contain; }`. [This way](https://jsfiddle.net/6yv230xp/) works as expected on Chrome canary 110 but doesn't work on Firefox nightly 109.
> 
> I tested it on Chrome Canary 110.0.5453.0, it does still hand off.

Sorry, my examination was incomplete. On Chrome, scrolling by mouse wheel or two-finger swipe doesn't hand off but scrolling by Arrow/PageUp/PageDown key does hand off.

> I found there's [an open spec issue](https://github.com/whatwg/html/issues/7732) for such case.

Thank you for letting me know!

**Comment by xPaw** [comment]
> I think it's helpful if scrolling on an overflow: auto element with no scroll range doesn't hand off to outer elements. The major use case is to prevent background scrolling when a modal dialog is open.

I agree with this, `overscroll-behavior` is much less useful in my use cases even when `overflow-y: scroll` is set, but the element is not scrollable because it fits on the screen, which causes overscroll to still happen.

**Comment by hiikezoe** [comment]
Chrome is changing their behavior; https://groups.google.com/a/chromium.org/g/blink-dev/c/OSq1yVSSdDE, I think it will be shipped in Chrome 144.

Also we (Mozilla) are going to change our behavior in https://bugzilla.mozilla.org/show_bug.cgi?id=1837436 soon.

So I am closing this issue.
